portability-core/src/main/java/org/dataportabilityproject/job/JobDao.java

```diff
@@ -182,12 +175,18 @@ public void updateJobStateToAssigneWithAuthData(String id, String encryptedExpor
         .setEncryptedExportAuthData(encryptedExportAuthData)
         .setEncryptedImportAuthData(encryptedImportAuthData)
         .build();
+    updateJobState(updatedJob, JobState.ASSIGNED_WITHOUT_AUTH_DATA,
+        JobState.ASSIGNED_WITH_AUTH_DATA);
+  }
 
-    updateJob(updatedJob, JobState.ASSIGNED_WITH_AUTH_DATA);
-    // Delete the job in the previous state
-    String previousKey = createKey(JobState.ASSIGNED_WITHOUT_AUTH_DATA, id);
-    storage.delete(
-        previousKey); // TODO: Determine whether to Tombstone or Delete from unassigned once it's entered in ready state
+  /**
+   * Deletes an existing {@code job} in storage with the given {@code jobState}.
+   */
+  public void deleteJob(String id, JobState jobState) throws IOException {
```

Comment:
It looks like deleteJob is only called in testing - is this something that we can expose for TestOnly?

Reply:
I think we'll eventually need it when implementing the restart/recovery parts of the lifecycle

Reply:
Oh right, that makes sense! Thanks!